### ISSUE TITLE
[a11y]: Condition Builder  accessibility violations part 1

### ISSUE THREAD

**Issue by amal-k-joy** [carbon-design-system/ibm-products#6788]
These are issues taken from the list of issues shared by Raghu in his accessibility testing

- [ ] Cancelling /Closing of the Modals with in the Condition Building process (issue no 7):
             1. Modals are not given with Cancel /Close Controls .
              2. Escape key is not Closing the Modal consistently - For example - Of the Focus is on Search Edit Field - pressing Escape is not 
     Closing the Modal.
**Solution**: Clear the text inside the search box on escape key press if any search text is present. Close the popover on escape only if search box is empty. If some search text is present, user has to press escape twice to close the popover.

- [ ] Submit Button not provided , unable to submit  after selecting from  "Multi select List Box " (issue no 9):
           Users are unable to submit the modal due to the absence of a submit button (the Enter key does not work for submission). When the operator "Is one of" is selected, a multi-select list box appears for choosing multiple options, but there is no way to submit the form afterward.

  **Solution**: We can close/submit the multiselect popover when enter is pressed . Space can be used to toggle between the selection. Implement this on every other input types. On a long text input, press enter will close popover with entered value. Shift + enter will enable user to move to new line inside the text area.

- [ ] Hierarchy of Heading Level provided for the Main Header of the     Modals with in the Condition Building process (issue no 5).
Main Heading of the Modal is Not given with right Heading level 
**Solution**: Replace with carbon Heading component.
